src/Resources/contao/widgets/ComponentStyleSelect.php

```diff
@@ -138,10 +138,24 @@ public function generate()
             $opts = StringUtil::deserialize($objStyleGroups->cssClasses);
 
             foreach ($opts as $opt) {
-                $arrFieldOptions[] = array(
+                $option = array(
                     'label' => $opt['value'] ?: $opt['key'],
                     'value' => $opt['key']
                 );
+
+                // skip third-party fields
+                if (isset($GLOBALS['TL_HOOKS']['styleManagerGetDynamicFieldOptions']) && \is_array($GLOBALS['TL_HOOKS']['styleManagerGetDynamicFieldOptions']))
```

Comment:
The names of the hooks in the StyleManager always tell the current position of the context. Here we should rather name the hook like this: `styleManagerGroupFieldOptions`.

Reply:
Done in 57a7ab1